sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveShim.scala

```diff
@@ -600,11 +600,14 @@ private[client] class Shim_v0_13 extends Shim_v0_12 {
             .asInstanceOf[JArrayList[Partition]]
         } catch {
           case ex: InvocationTargetException if ex.getCause.isInstanceOf[MetaException] &&
-              !tryDirectSql =>
-            logWarning("Caught Hive MetaException attempting to get partition metadata by " +
-              "filter from Hive. Falling back to fetching all partition metadata, which will " +
-              "degrade performance. Modifying your Hive metastore configuration to set " +
-              s"${tryDirectSqlConfVar.varname} to true may resolve this problem.", ex)
+              (!tryDirectSql || ex.getCause.getMessage.contains(
```

Comment:
In hindsight, I believe this approach of testing the hive configuration may be fundamentally flawed. When testing the value of `HiveConf.ConfVars.METASTORE_TRY_DIRECT_SQL`, we need to know the value set in the Hive Metastore server. What we're testing here is the setting in this Hive client. They won't necessarily be the same, so we can get false positives and false negatives of this test here.